src/MachineKey.cs

```diff
@@ -113,7 +116,10 @@ public static byte[] Unprotect(byte[] protectedData, string validationKey, strin
                     // These KeyedHashAlgorithm instances are single-use; we wrap it in a 'using' block.
                     using (KeyedHashAlgorithm validationAlgorithm = CryptoConfig.CreateFromName(validationAlgorithmName) as KeyedHashAlgorithm)
```

Comment:
Do you not need to change these to `HashAlgorithm` as well?

If it is not a `KeyedHashAlgorithm` it will still end up with a null ref the next time `validationAlgorithm` is used.

Reply:
oh..I'm sorry. I had two copy of code. I pushed incomplete one. I have re pushed the missing code, plz check again.